src/cascadia/TerminalSettings/KeyChord.cpp

```diff
@@ -39,4 +115,171 @@ namespace winrt::Microsoft::Terminal::Settings::implementation
+    // - hstr: the string to parse into a keychord.
+    // Return Value:
+    // - a newly constructed KeyChord
+    winrt::Microsoft::Terminal::Settings::KeyChord KeyChord::FromString(const hstring& hstr)
+    {
+        std::wstring wstr{ hstr };
+
+        // Split the string on '+'
+        std::wstring temp;
+        std::vector<std::wstring> parts;
+        std::wstringstream wss(wstr);
+
+        while(std::getline(wss, temp, L'+'))
+        {
+            parts.push_back(temp);
+        }
+
+        // If we have > 4, something's wrong.
+        if (parts.size() > 4)
+        {
+            throw hresult_invalid_argument();
+        }
+
+        KeyModifiers modifiers = KeyModifiers::None;
+        int32_t vkey = 0;
+
+        // Look for ctrl, shift, alt. Anything else might be a key
+        for (const auto& part : parts)
+        {
+            if (part == CTRL_KEY)
+            {
+                modifiers |= KeyModifiers::Ctrl;
+            }
+            else if (part == ALT_KEY)
+            {
+                modifiers |= KeyModifiers::Alt;
+            }
+            else if (part == SHIFT_KEY)
+            {
+                modifiers |= KeyModifiers::Shift;
+            }
+            else
+            {
+                bool foundKey = false;
+                // For potential keys, look through the pairs of strings and vkeys
+                if (part.size() == 1)
+                {
+                    const wchar_t wch = part[0];
+                    // Quick lookup: ranges of vkeys that correlate directly to a key.
+                    if (wch >= L'0' && wch <= L'9')
+                    {
+                        vkey = static_cast<int32_t>(wch);
+                        foundKey = true;
+                    }
+                    else if (wch >= L'a' && wch <= L'z')
+                    {
+                        // subtract 0x20 to shift to uppercase
+                        vkey = static_cast<int32_t>(wch - 0x20);
+                        foundKey = true;
+                    }
+                    else if (wch >= L'A' && wch <= L'Z')
+                    {
+                        vkey = static_cast<int32_t>(wch);
+                        foundKey = true;
+                    }
+                }
+
+                // If we didn't find the key with a quick lookup, search the
+                // table to see if we have a matching name.
+                if (!foundKey)
+                {
+                    for (const auto& pair : vkeyNamePairs)
+                    {
+                        if (pair.second == part)
+                        {
+                            vkey = pair.first;
+                            foundKey = true;
+                            break;
+                        }
+                    }
+                }
+
+                // If we weren't able to find a match, throw an exception.
+                if (!foundKey)
+                {
+                    throw hresult_invalid_argument();
```

Comment:
Do we also want an error dialogue box like in [#622](https://github.com/microsoft/terminal/issues/622)? Or maybe just ignore it? I'm ok with this behavior for now but we should probably decide on behavior and file it as a separate issue for after.

Reply:
Yea probably, but I'll make that a separate task for once both this and 622 are merged.

Reply:
Here's a concern: should not being able to deserialize a single key binding blast the entire profiles into outer space? even with an error message, that sucks

Reply:
Yea, I don't think it should. I'd think that failing to parse a single key would still ignore the bad binding, but display the error (somehow)